src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -451,6 +451,50 @@ private function isSimpleInstance($id, Definition $definition)
     }
 
     /**
+     * Checks if the definition is a trivial instance.
+     *
+     * @param Definition $definition
+     *
+     * @return bool
+     */
+    private function isTrivialInstance(Definition $definition)
```

Comment:
actually, as a private method, this *is* an internal detail :)
more importantly, the "trivial" is important here. It's not only that the instantiation can be inlined, it's also that the instantiation is trivial, this meaning that it mostly a simple `new Foo()`.
More complex instantiation would also be inlineable (eg `Factory::create(new Foo($this->get('bar')))`), but we don't want to inline those (we need a limit, being trivial is the one, for what it means (it only means whatever logic is in the method, see patch :) .)